src/EditDistEpochRecordFilter.hpp

```diff
@@ -82,22 +82,65 @@ namespace geopm
             ///        Exception with GEOPM_ERROR_INVALID type.
             static void parse_name(const std::string &name,
                                    int &history_buffer_size,
-                                   int &min_stable_period,
+                                   int &min_hysteresis_base_period,
                                    double &stable_period_hysteresis,
                                    double &unstable_period_hysteresis);
         private:
+            /// Implements:
+            ///  1. The stable period detector state machine,
+            ///  2. Returns True if the last record passed to m_edpd
+            ///     is considered as an epoch marker. Marked
+            ///     records are expected to be spaced (in number of
+            ///     records) the length of a period plus/minus erroneous
+            ///     records.
+            ///
+            /// STATE MACHINE OPERATION
+            /// -------------------------
+            /// PERIOD_DETECTED state means we are observing a stable period of N and as long as we stay in this
+            /// state we expect to observe an epoch distance of length N plus/minus erroneously added/removed
+            /// calls, which are indicated by the edit distance (score).
+            ///
+            /// The period stability is measured by comparing the period detected with the most current record vs
+            /// detected with the previous. We store the previous in m_last_period.
+            ///
+            /// The conditions for NO_PERIOD_DETECTED -> PERIOD_DETECTED state transition:
+            ///    * Stable period of N detected by String Edit Distance algorithm for
+            ///      the last MAX(N, min_hysteresis_base_period) x STABLE_PERIOD_HYSTERESIS records.
+            ///    * Only periods >= MIN_DETECTABLE_PERIOD are considered as potentially stable.
+            ///
+            /// The conditions for PERIOD_DETECTED -> NO_PERIOD_DETECTED state transition:
+            ///    * Detected period deviated from N for the last UNSTABLE_PERIOD_HYSTERESIS x N records.
+            ///
+            /// The conditions for reporting a record as epoch marker:
+            ///    * "N plus/minus erroneous records" passed since last reported epoch marker.
             bool epoch_detected();
 
+            /// The String Edit Distance algorithm that finds the
+            /// patterns are implemented in this object.
             std::shared_ptr<EditDistPeriodicityDetector> m_edpd;
-            const int m_min_stable_period;
+            // Parameter for the epoch detection algorithm. See
```

Comment:
Are these intended to be doxygen?  /// required?

Reply:
Intention is not to have doxygen for the private variables but just to make the code more readable.